Resources/public/js/views/ez-usermenuview.js

```diff
@@ -0,0 +1,225 @@
+     */
+    Y.namespace('eZ');
+
+    var CLASS_HIDDEN = 'is-user-menu-hidden',
+        CLASS_USER_PROFILE = 'ez-user-profile',
+        SELECTOR_USER_PROFILE = '.' + CLASS_USER_PROFILE,
+        SELECTOR_USER_MENU = '.ez-user-menu';
+
+    /**
+     * The User Menu view
+     *
+     * @namespace eZ
+     * @class UserMenuView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.UserMenuView = Y.Base.create('userMenuView', Y.eZ.TemplateBasedView, [], {
+        events: {
+            '.ez-user-menu-item': {
+                'tap': '_toggleUserMenu'
+            },
+            '.ez-user-menu-item-logout': {
+                'tap': '_logOut'
+            }
+        },
+
+        initializer: function () {
+            this.on('*:toggleUserMenu', this._toggleUserMenu);
+        },
+
+        /**
+         * Renders the user menu view
+         *
+         * @method render
+         * @return {eZ.UserMenuView} the view itself
+         */
+        render: function () {
+            var container = this.get('container');
+
+            container.setHTML(this.template(this.getAttrs()));
+            container.on('clickoutside', this._hideMenuOnClickOutside, this);
+
+            this._renderMenuItems();
+
+            return this;
+        },
+
+        /**
+         * Tap event handler on the logout link
+         *
+         * @method _logOut
+         * @protected
+         * @param event {Object} tap event facade
+         */
+        _logOut: function (event) {
+            event.preventDefault();
+            this.fire('logOut', {
+                originalEvent: event
+            });
+        },
+
+        /**
+         * Toggles the visibility of a user menu.
+         *
+         * @method _toggleUserMenu
+         * @protected
+         */
+        _toggleUserMenu: function () {
+            var methodName = this.get('displayed') ? 'addClass' : 'removeClass',
+                userMenu = this.get('container').one(SELECTOR_USER_MENU);
+
+            userMenu[methodName](CLASS_HIDDEN);
+            this.set('displayed', !this.get('displayed'));
+        },
+
+        /**
+         * Checks if clicked outside the user profile container and hides a user menu if it is visible.
+         *
+         * @method _hideMenuOnClickOutside
+         * @protected
+         */
+        _hideMenuOnClickOutside: function (event) {
+            if (!event.target.hasClass(CLASS_USER_PROFILE) && event.target.ancestor(SELECTOR_USER_PROFILE) === null && this.get('displayed')) {
+                 this._toggleUserMenu();
+            }
+        },
+
+        /**
+         * Renders menu items views
+         *
+         * @method _renderMenuItems
+         * @protected
+         */
+        _renderMenuItems: function () {
+            var menuItemsContainer = this.get('container').one(SELECTOR_USER_MENU),
+                logOutView = this.get('logOutItemView'),
+                additionalItems = this.get('additionalItems');
```

Comment:
I think `eZ.UserMenuView` should be simplified, IMHO it's a bit complicated partly because you do pretty much the same thing as for the navigation items while the constraints are a different. Also the way you handle the logout is not correct because the `eZ.UserMenuView` and the view for the logout link are too much coupled, `eZ.UserMenuView` should have no knowledge of what is generated by its sub-views.

So here is how this could be done:

1. I would put all the items view in one attribute (`items`) as the logout item is not really different from the others.
1. Sorting by an attribute before rendering is a good idea but we tend to use `priority` over `order` (ie big values first) and the base view class `eZ.UserMenuItemView` should have such attribute with a default value.
1. to ease the whole process, I would make sure the `eZ.UserMenuView` container is an `<ul>`, that way you can directly add the sub-views to the container.
1. the trick where you describe the sub-views with a plain object with the `Constructor` property is not necessary here, so you should remove that option, the `items` attribute will handle only view instances.
1. as explained above, the `cssClassSuffix` system tends to couple a bit the components together. For the items in the user menu, one use case is to fire an event when the user clicks on the items like for logout. So I think you should implement a `eZ.UserMenuItemFireEventView` (feel free to propose a better name :)) which extends `eZ.UserMenuItemView` and expects in addition to the title an event name that is triggered when the user clicks on it.

Basically, you'll end up having the `items` attribute defined like:

```js
items: {
    readOnly: true,
    valueFn: function () {
        return [
            new Y.eZ.UserMenuItemFireEventView({
                bubbleTargets: this,
                title: "Logout",
                eventName: 'logout',
            }),
        ];
    },
}
```
for this to work, you'll have to make the `active` flag is correctly updated on the items. That means you have to implement an `activeChange` event handler so that the items receive the correct value when the active flag changes.